### PR TITLE
add_fail_test function

### LINKED ISSUE #29: fail-tests (and warnings)
Eventually mrdox will have to emit errors of some kind for example if a javadoc is malformed or if there are duplicate javadoc definitions. For this we will need fail-tests.

We might also want warnings, such as when parameters are not documented in the source inputs.

### PR DESCRIPTION
**Pull request by klemens-morgenstern**
Closes #29

### CODE REVIEW

File: tests/FailTest.cmake [hunk 1]
```diff
@@ -0,0 +1,37 @@
+if(WIN32)
+    set(MRDOX_FT_RUNNER ${CMAKE_CURRENT_LIST_DIR}/fail_test.bat)
+else()
+    set(MRDOX_FT_RUNNER ${CMAKE_CURRENT_LIST_DIR}/fail_test.sh)
```
Comment (vinniefalco) [inline tests/FailTest.cmake:4]: dafuq is "CMAKE_CURRENT_LIST_DIR" ? what is a listfile?

Comment (klemens-morgenstern) [reply tests/FailTest.cmake:4]: The directory of the current file

**Comment by vinniefalco** [comment]
Is that a merge commit?

**Comment by vinniefalco** [comment]
This needs to be rebased, and the scripts going to the right place (tools/ or scripts/)

**Pull request closed without merging**